substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1826,13 +1824,30 @@ impl<T: Config> Pallet<T> {
 			"VoterList contains non-staker"
 		);
 
+		Self::check_payees()?;
 		Self::check_nominators()?;
 		Self::check_exposures()?;
 		Self::check_paged_exposures()?;
 		Self::check_ledgers()?;
 		Self::check_count()
 	}
 
+	/// Invariants:
+	/// * A bonded ledger should always have an assigned `Payee`.
+	/// * The number of entries in `Payee` and of bonded staking ledgers *must* match.
+	fn check_payees() -> Result<(), TryRuntimeError> {
+		for (acc, _) in Ledger::<T>::iter() {
+			ensure!(Payee::<T>::get(acc).is_some(), "bonded ledger does not have payee set");
```

Comment:
You should add some logging here so that we are able to find out which account is the problematic one?